core/src/com/cloud/agent/resource/virtualnetwork/facade/AbstractConfigItemFacade.java

```diff
@@ -104,13 +109,25 @@ public static AbstractConfigItemFacade getInstance(final Class<? extends Network
         return instance;
     }
 
+
+    private static String appendUuidToJsonFiles(final String filename) {
+        String remoteFileName = new String(filename);
+        if (remoteFileName.endsWith("json")) {
+            remoteFileName += "." + UUID.randomUUID().toString();
+        }
+        return remoteFileName;
+    }
+
     protected List<ConfigItem> generateConfigItems(final ConfigBase configuration) {
         final List<ConfigItem> cfg = new LinkedList<>();
 
-        final ConfigItem configFile = new FileConfigItem(VRScripts.CONFIG_PERSIST_LOCATION, destinationFile, gson.toJson(configuration));
+        final String remoteFilename = appendUuidToJsonFiles(destinationFile);
+        s_logger.debug("Transformed filename " + destinationFile + " to " + remoteFilename);
```

Comment:
Please wrap in a ``if (s_logger.isDebugEnabled())`` block to avoid unnecessary string creation due to pre-method string concatenation when ``DEBUG`` is not enabled.